public/app/plugins/datasource/loki/result_transformer.ts

```diff
@@ -447,7 +447,7 @@ export function processRangeQueryResponse(
   switch (response.data.resultType) {
     case LokiResultType.Stream:
       return of({
-        data: lokiStreamsToDataframes(response.data.result, target, limit, config, reverse),
+        data: lokiStreamsToDataframes(response.data.result.slice(0, limit), target, limit, config, reverse),
```

Comment:
@Estrax can I ask what is the reason for this change? Does it have anything to do with the explore query editor limit?  Just curious.

Reply:
Yes - line limit equal to 0 or -1 is technically correct and some datasources return an entire table or document then, so working on a slice is one of the simplest workarounds to that problem.

Reply:
Makes sense! Thanks for explanation. 🙏